pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -77,12 +80,87 @@ func (r *AtlasDeploymentReconciler) ensureAdvancedDeploymentState(ctx *workflow.
 	}
 }
 
+func handleSearchNodes(ctx *workflow.Context, deployment *akov2.AtlasDeployment, projectID string) workflow.Result {
+	// Get the current state from Atlas
+	// If no nodes configured in atlas, but some in the operator - create them. If not - do nothing
+	// If nodes already configured in atlas and in the operator - update them. If not - delete them
+	ctx.Log.Debug("starting search node processing")
+	defer ctx.Log.Debug("finished search node processing")
+
+	nodesInAkoEmpty := deployment.Spec.DeploymentSpec.SearchNodes == nil || len(deployment.Spec.DeploymentSpec.SearchNodes) == 0
+
+	nodesInAtlasEmpty := false
+	currentNodesInAtlas, httpResp, err := ctx.SdkClient.AtlasSearchApi.GetAtlasSearchDeployment(ctx.Context, projectID, deployment.GetDeploymentName()).Execute()
+	if err != nil {
+		if httpResp == nil {
+			ctx.Log.Debugf("unable to get current search nodes status: %v", err)
+			return workflow.Terminate(workflow.SearchNodesReady, err.Error())
+		}
+
+		switch httpResp.StatusCode {
+		// TODO: Should be only NotFound: CLOUDP-239015
+		case http.StatusBadRequest, http.StatusNotFound:
+			ctx.Log.Debug("no search nodes in atlas found")
+			nodesInAtlasEmpty = true
+		default:
+			return workflow.Terminate(workflow.SearchNodesReady, err.Error())
+		}
+	}
+
+	if !nodesInAtlasEmpty && currentNodesInAtlas.GetStateName() != "IDLE" {
+		msg := fmt.Sprintf("search nodes are not ready: %v", currentNodesInAtlas.GetStateName())
+		ctx.Log.Debug(msg)
+		return workflow.InProgress(workflow.SearchNodesReady, msg)
```

Comment:
what if `nodesInAkoEmpty` is `true` in this case?

Reply:
Moved this down so that we check if the nodes are idle only if the AKO and Atlas specs are the same.